hkm/views/views.py

```diff
@@ -521,21 +522,70 @@ def handle_search(self, request, *args, **kwargs):
         if self.date_facets:
             facets['main_date_str'] = self.date_facets
         load_all_pages = bool(int(request.GET.get('loadallpages', 1)))
-        records = []
+
+        session_search_result = copy.deepcopy(request.session.get('search_result', {}))
+        self.search_result = session_search_result
+        records = session_search_result.get('records', []) if self.search_term != request.GET.get('search', '') else []
+
+        search_term_changed = self.search_term != request.session.get('search')
+        page_changed = self.page != request.session.get('page')
+        author_facets_changed = len(self.author_facets) != len(request.session.get('author_facets', []))
+        date_facets_changed = len(self.date_facets) != len(request.session.get('date_facets', []))
```

Comment:
So it is enough to check if the number of facets has changed, it is not necessary to compare the values of those?

Reply:
Every time author or date is changed, search is executed again automatically. So checking only the length should be enough.